app/models/binary/DataStore.scala

```diff
@@ -84,6 +102,39 @@ class DataStoreService @Inject()(dataStoreDAO: DataStoreDAO)(implicit ec: Execut
       result <- block(dataStore)
     } yield result).getOrElse(Forbidden(Json.obj("granted" -> false, "msg" -> Messages("dataStore.notFound"))))
 
+  def createNewUserDataStore(config: UserDataStoreConfig, organizationId: ObjectId): Fox[TemporaryFile] = {
+    def writeInstallScriptToFile(fileContent: String): TemporaryFile = {
+      val file = temporaryFileCreator.create()
+      val fw = new FileWriter(file)
+      fw.write(fileContent)
+      fw.close()
+      file
+    }
+
+    for {
+      key <- new CompactRandomIDGenerator().generate
+      organization <- organizationDAO.findOne(organizationId)(GlobalAccessContext)
+      (_, assets) <- checkForUpdate()
+      jarURL = getUrlForFileEnding(assets, ".jar")
+      updateScriptURL = getUrlForFileEnding(assets, ".sh")
+      _ <- dataStoreDAO.insertOne(DataStore(
+        config.name,
+        config.url,
+        config.url,
+        key,
+        onlyAllowedOrganization = Some(organizationId))) ?~> "dataStore.create.failed"
+      installScript = InstallScript.getInstallScript(config.name,
+                                                     config.url,
+                                                     config.port,
+                                                     organization.name,
+                                                     key,
+                                                     conf.Http.uri,
+                                                     updateScriptURL,
+                                                     jarURL)
+
+    } yield writeInstallScriptToFile(installScript)
```

Comment:
Is there a way to remove it again? Or will the user after disabling their local datastore forever see the errors when hitting the dashboard dataset refresh button?

Reply:
Yes, now there is. Have a look at https://github.com/scalableminds/webknossos/pull/5579/files#diff-602b1fba87abc9f99dcd52e9d8de29dc2d6b5b7f444c4db7b39920ffb2c11130R96-R109